rules/math_big_rat.go

```diff
@@ -0,0 +1,98 @@
+package rules
+
+import (
+	"fmt"
+	"go/ast"
+
+	"github.com/securego/gosec/v2"
+)
+
+type usingOldMathBig struct {
+	gosec.MetaData
+}
+
+func (r *usingOldMathBig) ID() string {
+	return r.MetaData.ID
+}
+
+func getXName(elem ast.Stmt) (*ast.IfStmt, string, error) {
+	ifStmt, ok := elem.(*ast.IfStmt)
+	if !ok {
+		return nil, "", fmt.Errorf("not an if")
+	}
+	ifCond, ok := ifStmt.Cond.(*ast.BinaryExpr)
+	if !ok {
+		return nil, "", fmt.Errorf("not a BinaryExpr")
+	}
+	ifCondX, ok := ifCond.X.(*ast.Ident)
+	if !ok {
+		return nil, "", fmt.Errorf("X not an Ident")
+	}
+	return ifStmt, ifCondX.Name, nil
+}
+
+func (r *usingOldMathBig) Match(node ast.Node, ctx *gosec.Context) (gi *gosec.Issue, err error) {
+	callExpr, callExprOk := node.(*ast.CallExpr)
+	if !callExprOk {
+		return nil, nil
+	}
+
+	packageName, callName, callInfoOk := gosec.GetCallInfo(callExpr, ctx)
+	if callInfoOk != nil {
+		return nil, nil
+	}
+
+	if packageName != "math/big.Rat" || callName != "SetString" {
+		return nil, nil
+	}
+
+	funcDecl, funcDeclErr := gosec.FindFuncDecl(callExpr, ctx)
+	if funcDeclErr != nil {
+		return nil, nil
+	}
+
+	for _, elem := range funcDecl.Body.List {
+		firstIf, firstName, err := getXName(elem)
```

Comment:
You are introspecting the function and make assumptions about the code. This is completely not scaling as soon as the Go team  will change the function code. gosec doesn't do anything like this in any rule. 

Looking at the patch https://go-review.googlesource.com/c/go/+/379537/. I think you can just simplify the check a lot by only checking the type of the variable and in the warning message just mention the vulnerable Go version and CVE number.

The SetString on big.Rat is really specific method, I don't expect that is used in a lot of projects.

Reply:
Do you propose to issue regardless of the version of Go? Or still check the version using the runtime library?

Reply:
Yes, I was talking about the same library

Reply:
I create new PR #819, because this one is closed, and I can't reopen it